docs/DeveloperGuide.md

```diff
@@ -1,34 +1,426 @@
+With this document, you should have a better understanding of the framework of the application. 
 
-### Value proposition
+### 1.2. Target Reader
 
-{Describe the value proposition: what problem does it solve?}
+The target reader of this developer guide are:
 
-## User Stories
+- Project Managers
+- Future Developers
+- Build Verification Testers
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+### 1.3. Brief Description
 
-## Non-Functional Requirements
+OrgaNice! is a task manager integrated with a Study Area search function. You can use it to manage your tasks, be it events or deadlines. On top of that, 
+it is capable of assisting students in finding Study Areas that meets their desired criteria.
 
-{Give non-functional requirements}
+### 1.4. Main Features
 
-## Glossary
+1. Scheduling Tasks
+    - You can manage tasks based on priority for events and countdown for deadlines.
+1. Study Area. For more information of the design and implementation for this feature, click [here](#31-scheduling-tasks)
+    - You can find a Study Area that meets your criteria. The software however have a limited number of supported 
+    criteria. For more information of the design and implementation for this feature, click [here](#32-listing-study-areas)
+1. Notes
+	- You can enter notes based on school modules. Notes support undo and redo operations.
 
-* *glossary item* - Definition
+## 2. Design   
+ {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}    
 
-## Instructions for Manual Testing
+[comment]: # (@@author GanapathySanathBalaji)
+### 2.1. Architecture  
+  
+![Architecture](images/Architecture.png)
+<div>Figure 1. Overall Architecture of OrgaNice!</div>
+<br>
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+1. Duke - Main component which controls the flow of execution.
+
+1. Ui - Component used to get input from the user and display results on the monitor.
+
+1. Parser - Component used to abstract out the command based on user's input, so that the command can be executed later.
+
+1. Command - Component contains information and implementation on how to execute various types of commands.
+
+1. Task - Component contains details about handling the task list and related operations.
+
+1. StudyArea - Component contains details about handling queries for study area search.
+
+1. Notes - Component contains details about Notes related operations.
+
+1. Exception - Component contains the various types of exceptions encountered when OrgaNice! is run.
+
+1. ResourceLoader - Component handles loading and saving of the task list and study area details to local storage.
```

Comment:
I really appreciate this short portion, at a quick glance I'm able to understand what the core classes are purposed for. This will help new developers get up to speed fast, coupled together with the object diagram.